### DISCUSSION TITLE
[DNSDIST] Send query to different servers based on source address

### DISCUSSION THREAD

**Discussion by SuperTrollMan** [PowerDNS/pdns#14928] [Q&A]
I'm a new dnsdist user, and so far it seems intuitive and easy to use.
However I'd like to be able to send the DNS requests to different servers,

Example *IP's have been substituted for RFC1918 addresses to simplify the topology*
Client 1 (192.168.0.10) > dnsdist server > 1.1.1.1
Client 2 (192.168.0.20) > dnsdist server > 1.1.1.3

The only useful search I've found is this:
https://serverfault.com/questions/732976/powerdns-is-it-possible-to-send-different-results-based-on-server-ip-used-by-th
Which suggests it would be possible with LUA using dnsdist.

Is this possible with dnsdist, and would you be able to point me in the right direction?

Secondly, if it is possible, can you also specify an internal zone in the mix too?

Example *IP's have been substituted for RFC1918 addresses to simplify the topology*
Client 1 (192.168.0.10)> lookup google.com > dnsdist server > 1.1.1.1
Client 1 (192.168.0.10)> lookup mydomain.local > dnsdist server > 192.168.10.10
Client 2 (192.168.0.20) > lookup google.com > dnsdist server > 1.1.1.3

Thanks

**Comment by rgacogne** [comment]
It can be done by using built-in rules:
- NetmaskGroupRule [1] can be used to filter on the provenance
- QNameSuffixRule [2] can be used to filter on the requested name
- AndRule [3] and OrRule [4] can be used to combine other rules
- PoolAction [5] can route queries to a given pool of backends 

So you can put your backends in separate pools [6], and route queries to pools according to your needs.

[1]: https://dnsdist.org/reference/selectors.html?highlight=netmaskgrouprule#NetmaskGroupRule
[2]: https://dnsdist.org/reference/selectors.html?highlight=suffixmatch#QNameSuffixRule
[3]: https://dnsdist.org/reference/selectors.html#AndRule
[4]: https://dnsdist.org/reference/selectors.html#OrRule
[5]: https://dnsdist.org/reference/actions.html?highlight=poolaction#PoolAction
[6]: https://dnsdist.org/guides/serverpools.html
